### PR TITLE
Ss.refresh for all tabs

### PR DESCRIPTION
**Pull request by swetha571**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

# Description of the change
Refresh for all tabs
...

# Motivation for the change

...

### CODE REVIEW

File: src/components/pages/dashboard/dashboard.js [hunk 1]
```diff
@@ -66,6 +73,14 @@ class DashboardPage extends Component {
     );
   }
 
+  refreshData() {
+      this.setState({
```
Comment (spryor) [inline src/components/pages/dashboard/dashboard.js]: make set state one line

File: src/components/pages/devices/devices.js [hunk 2]
```diff
@@ -26,8 +27,23 @@ class DevicesPage extends Component {
...
+      lastRefreshed: new Date(),
       contextBtns: ''
     };
+    this.refreshData = this.refreshData.bind(this);
+  }
+
+  refreshData() {
+      this.setState({ lastRefreshed: new Date() },
+      ()=> this.props.actions.loadDevices(true)
+  ); }
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: Closing the method and closing the setState should be on separate lines.

File: src/components/pages/devices/devices.js [hunk 3]
```diff
@@ -63,11 +79,16 @@ class DevicesPage extends Component {
...
-          <PcsBtn svg={AddSvg} 
-                  onClick={this.props.openProvisionFlyout} 
+          <PcsBtn svg={AddSvg}
+                  onClick={this.props.openProvisionFlyout}
                   value={lang.PROVISIONDEVICES} />
         </ContextFilters>
-        <PageContent>
+        <PageContent className="devices-grid-container">
+        <div className="timerange-selection">
+          <span className="last-refreshed-text"> {`${lang.LAST_REFRESHED} | `} </span>
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: There spaces after and before the `><`, is that intended to add whitespace to the page? If not please remove the spaces.

File: src/components/pages/devices/devices.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    .last-refreshed-time {
+      margin-right: 10px;
+      color: #fff;
+    }
+
+    .last-refreshed-text {
+      margin-left: 10px;
+      margin-right: 6px;
+      color: #afb9c3;
```
Comment (spryor) [inline src/components/pages/devices/devices.scss]: Import `_colors.scss` here and use` $lightestGray`

File: src/components/pages/devices/devices.scss [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    .refresh-icon {
+      background: url("../../../assets/icons/Refresh.svg") no-repeat center;
```
Comment (spryor) [inline src/components/pages/devices/devices.scss]: Why is this not an actual page asset?

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 6]
```diff
@@ -48,14 +49,13 @@ class RulesAndActionsPage extends Component {
...
+    this.refreshData = this.refreshData.bind(this);
   }
 
   componentDidMount() {
-    ApiService.getRuleList()
-      .then(({ Items }) => this.setState({ 
-        rulesAndActions: Items, 
-        showSpinner: false 
-      }));
+     this.setState({ showSpinner: false, lastRefreshed: new Date() },
```
Comment (spryor) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: Make the first parameter a new line as well.

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 7]
```diff
@@ -166,6 +166,12 @@ class RulesAndActionsPage extends Component {
     }
   };
 
+  refreshData() {
+    this.setState({ lastRefreshed: new Date(), showSpinner: false },
```
Comment (spryor) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: Same comment about the first parameter

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 8]
```diff
@@ -166,6 +166,12 @@ class RulesAndActionsPage extends Component {
...
+    ()=> this.props.actions.loadRulesList()
```
Comment (spryor) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: Also check the indentation, Every file should 2 space tabs.

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 9]
```diff
@@ -200,25 +206,30 @@ class RulesAndActionsPage extends Component {
...
     };
 
     return (
       <PageContainer>
         <TopNav breadcrumbs={'Rules and Actions'} projectName={lang.AZUREPROJECTNAME} />
         {this.renderContextFilters()}
-        <PageContent>
+        <PageContent className="rules-and-actions-conatiner">
+        <div className="timerange-selection">
+          <span className="last-refreshed-text"> {`${lang.LAST_REFRESHED} | `} </span>
```
Comment (spryor) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: same question about the spaces around the `><`

File: src/components/pages/rulesAndActions/rulesAndActions.scss [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    
+  .last-refreshed-time {
+    margin-right: 10px;
+    color: #fff;
+  }
+
+  .last-refreshed-text {
+    margin-left: 10px;
+    margin-right: 6px;
+    color: #afb9c3;
```
Comment (spryor) [inline src/components/pages/rulesAndActions/rulesAndActions.scss]: color comment again

File: src/actions/kpiActions.js [hunk 11]
```diff
@@ -30,10 +30,21 @@ export const refreshAllChartData = (
...
+        firstDurationFrom = currentState.kpiReducer.firstDurationFrom;
+        firstDurationTo = currentState.kpiReducer.firstDurationTo;
+        secondDurationFrom = currentState.kpiReducer.secondDurationFrom;
+        secondDurationTo = currentState.kpiReducer.secondDurationTo;
+    }
     const devices = currentState.deviceReducer.devices;
     const deviceIdsCsv = (devices ? devices.items : []).map(({ Id })=> Id).join(',');
     dispatch(indicatorStart(refreshFlag ? 'kpi' : 'kpiInitial'));
-    dispatch({ type: types.KPI_REFRESH_CHART_DATA_START });
+    dispatch({ type: types.KPI_REFRESH_CHART_DATA_START, payload : {
```
Comment (spryor) [inline src/actions/kpiActions.js]: `payload :` remove the space before `:`

File: src/components/pages/dashboard/dashboard.js [hunk 12]
```diff
@@ -66,6 +72,12 @@ class DashboardPage extends Component {
     );
   }
 
+  refreshData() {
+    this.setState({ lastRefreshed: new Date() },
```
Comment (spryor) [inline src/components/pages/dashboard/dashboard.js]: move the first parameter to its own line

File: src/components/pages/devices/devices.js [hunk 13]
```diff
@@ -26,8 +27,24 @@ class DevicesPage extends Component {
...
     this.state = {
       softSelectedDeviceId: '',
+      lastRefreshed: new Date(),
       contextBtns: ''
     };
+    this.refreshData = this.refreshData.bind(this);
+  }
+
+  refreshData() {
+    this.setState({ lastRefreshed: new Date() },
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: move first parameter to new line

File: src/components/pages/devices/devices.js [hunk 14]
```diff
@@ -63,11 +80,16 @@ class DevicesPage extends Component {
...
+          <PcsBtn svg={AddSvg}
+                  onClick={this.props.openProvisionFlyout}
                   value={lang.PROVISIONDEVICES} />
         </ContextFilters>
-        <PageContent>
+        <PageContent className="devices-grid-container">
+        <div className="timerange-selection">
+          <span className="last-refreshed-text">{`${lang.LAST_REFRESHED} | `}</span>
+          <div className="last-refreshed-time">{this.state.lastRefreshed.toLocaleString()}</div>
+          <div onClick={this.refreshData} className="refresh-icon icon-sm"/>
```
Comment (spryor) [inline src/components/pages/devices/devices.js]: "[space]/>

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 15]
```diff
@@ -48,14 +48,19 @@ class RulesAndActionsPage extends Component {
...
   componentDidMount() {
-    ApiService.getRuleList()
-      .then(({ Items }) => this.setState({ 
-        rulesAndActions: Items, 
-        showSpinner: false 
-      }));
+    this.props.actions.showingRulesPage();
+    this.setState(
+      { showSpinner: false, lastRefreshed: new Date() },
+      ()=> this.props.actions.loadRulesList()
```
Comment (maple10001) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: space before ```=>```

File: src/components/pages/rulesAndActions/rulesAndActions.js [hunk 16]
```diff
@@ -166,6 +171,13 @@ class RulesAndActionsPage extends Component {
     }
   };
 
+  refreshData() {
+    this.setState(
+      { lastRefreshed: new Date(), showSpinner: false },
+      ()=> this.props.actions.loadRulesList()
```
Comment (maple10001) [inline src/components/pages/rulesAndActions/rulesAndActions.js]: space before ```=>```

**Pull request merged**